specs/protocol/block_header.md

```diff
@@ -0,0 +1,24 @@
+# Block Header
+
+## Application Header
+
+name                 | type       | description
+---------------------|------------|-----------------------------------------------------------------------------------------------------
+`height`             | `uin64`    | Height of this block.
+`timestamp`          | `uin64`    | Time this block was created, in [TAI64](https://cr.yp.to/libtai/tai64.html) format.
+`txCount`            | `uint64`   | Number of [transaction](./tx_format.md#transaction)s in this block.
+`txRoot`             | `byte[32]` | [Merkle root](./cryptographic_primitives.md#binary-merkle-tree) of [transaction](./tx_format.md#transaction)s in this block.
+`outputMessagesCount`|`uint64`| Number of [output message](./tx_format.md#outputmessage)s in this block.
+`outputMessagesRoot` | `byte[32]` | [Merkle root](./cryptographic_primitives.md#binary-merkle-tree) of [output message](./tx_format.md#outputmessage)s in this block.
+
+## Consensus Header
```

Comment:
We should have different consensus headers based on the network configuration. For example, we won't need things like validatorSetRoot or votingPower for PoA, and a PoA sidechain contract will likely be a lot simpler as a consequence.

Reply:
It would also be good if the consensus header included enough info to be authenticated by the contract, rather than only allowing the PoA node to post blocks. This way anyone could post block headers to Eth.

Reply:
Fixed in [b395226](https://github.com/FuelLabs/fuel-specs/pull/405/commits/b395226ce46f3f1e1a5a431e58a6eea661e2dd25)